java.util.NoSuchElementException: SRCFG00011: Could not expand value platform.quarkus.native.builder-image in property quarkus.native.builder-image

### Describe the bug

After the update of Quarkus from 3.15.1 to 3.17.6 following error occurs when building an image:
```
 * What went wrong:
 Execution failed for task ':application-name:quarkusAppPartsBuild'.
 > There was a failure while executing work items
    > A failure occurred while executing io.quarkus.gradle.tasks.worker.BuildWorker
       > Configuration validation failed:
         	java.util.NoSuchElementException: SRCFG00011: Could not expand value platform.quarkus.native.builder-image in property quarkus.native.builder-image
```
There was no change in the code. Only the update of Quarkus.
I did some research and found this issue: #19139. But we are already using `implementation enforcedPlatform('io.quarkus.platform:quarkus-bom:3.17.6')`
We are using the `io.quarkus:quarkus-container-image-jib` extension with following configuration:
```
quarkus.container-image.build=true
quarkus.container-image.push=true
quarkus.container-image.registry=reqestyUrl
quarkus.container-image.group=imageName
quarkus.container-image.username=username
quarkus.container-image.password=password
```
We don't create native images.

There is a workaround: When we set the following property: `quarkus.native.builder-image=mandrel`, then the images can be build again.

### Expected behavior

1. When no native images are build, it should not be necessary to set properties which are needed only for building native images.
2. 'mandrel' is the default value for `quarkus.native.builder-image`. So it should not be necessary to set the default value.

### Actual behavior

Without the mentioned workaround we are no more able to build images.

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

17

### Quarkus version or git rev

3.17.6

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.12

### Additional information

```
Caused by: io.smallrye.config.ConfigValidationException: Configuration validation failed:
 	java.util.NoSuchElementException: SRCFG00011: Could not expand value platform.quarkus.native.builder-image in property quarkus.native.builder-image
 	at io.smallrye.config.SmallRyeConfig.buildMappings(SmallRyeConfig.java:139)
 	at io.smallrye.config.SmallRyeConfig.<init>(SmallRyeConfig.java:93)
 	at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:767)
 	at io.quarkus.deployment.configuration.BuildTimeConfigurationReader.initConfiguration(BuildTimeConfigurationReader.java:412)
 	at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:141)
 	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:107)
 	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:351)
 	at io.quarkus.runner.bootstrap.AugmentActionImpl.createProductionApplication(AugmentActionImpl.java:178)
 	at io.quarkus.gradle.tasks.worker.BuildWorker.execute(BuildWorker.java:62)
 	at app//org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
 	at app//org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:54)
 	at app//org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:48)
 	at app//org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:100)
 	at app//org.gradle.workers.internal.AbstractClassLoaderWorker.executeInClassLoader(AbstractClassLoaderWorker.java:48)
 	at app//org.gradle.workers.internal.IsolatedClassloaderWorker.run(IsolatedClassloaderWorker.java:49)
 	at app//org.gradle.workers.internal.IsolatedClassloaderWorker.run(IsolatedClassloaderWorker.java:30)
 	at app//org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:109)
 	at app//org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:78)
 	at app//org.gradle.process.internal.worker.request.WorkerAction$1.call(WorkerAction.java:159)
 	at app//org.gradle.process.internal.worker.child.WorkerLogEventListener.withWorkerLoggingProtocol(WorkerLogEventListener.java:41)
 	at app//org.gradle.process.internal.worker.request.WorkerAction.lambda$run$1(WorkerAction.java:156)
 	at app//org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:85)
 	at app//org.gradle.process.internal.worker.request.WorkerAction.run(WorkerAction.java:148)
 	at java.base@17.0.12/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 	at java.base@17.0.12/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
 	at java.base@17.0.12/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
 	at java.base@17.0.12/java.lang.reflect.Method.invoke(Unknown Source)
 	at app//org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
 	at app//org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
 	at app//org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
 	at app//org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
 	at app//org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
 	at app//org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
 	at app//org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48)
 	... 3 more
```

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/AbstractQuarkusExtension.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/AbstractQuarkusExtension.java
@@ -80,7 +80,7 @@ public abstract class AbstractQuarkusExtension {
 
         // Used to handle the (deprecated) buildNative and testNative tasks.
         project.getExtensions().getExtraProperties().getProperties().forEach((k, v) -> {
-            if (k.startsWith("quarkus.")) {
+            if (k.startsWith("quarkus.") || k.startsWith("platform.quarkus.")) {
                 forcedPropertiesProperty.put(k, v.toString());
             }
         });
@@ -126,7 +126,7 @@ public abstract class AbstractQuarkusExtension {
 
         // Used to handle the (deprecated) buildNative and testNative tasks.
         project.getExtensions().getExtraProperties().getProperties().forEach((k, v) -> {
-            if (k.startsWith("quarkus.")) {
+            if (k.startsWith("quarkus.") || k.startsWith("platform.quarkus.")) {
                 forcedPropertiesProperty.put(k, v.toString());
             }
         });

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusBuildTask.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusBuildTask.java
@@ -251,6 +251,9 @@ public abstract class QuarkusBuildTask extends QuarkusTask {
             for (String key : config.getMapKeys("quarkus").values()) {
                 values.put(key, config.getConfigValue(key).getValue());
             }
+            for (String key : config.getMapKeys("platform.quarkus").values()) {
+                values.put(key, config.getConfigValue(key).getValue());
+            }
             return values;
         });
 

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusTask.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusTask.java
@@ -19,7 +19,7 @@ import io.quarkus.gradle.extension.QuarkusPluginExtension;
 import io.quarkus.utilities.OS;
 
 public abstract class QuarkusTask extends DefaultTask {
-    private static final List<String> WORKER_BUILD_FORK_OPTIONS = List.of("quarkus.");
+    private static final List<String> WORKER_BUILD_FORK_OPTIONS = List.of("quarkus.", "platform.quarkus.");
 
     private final transient QuarkusPluginExtension extension;
     protected final File projectDir;

